### PR TITLE
Add tests for ABX VPP plugin

### PR DESCRIPTION
**Pull request by PantheonVladimirB**
Signed-off-by: Vladimir Bernolak <vladimir.bernolak@pantheon.tech>

### CODE REVIEW

File: vpp/abx/vpp2202/test/test_abx.py [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+from vpp_sub_interface import VppDot1QSubint
+from vpp_acl import AclRule, VppAcl, VppAclInterface
+
+from ipaddress import IPv4Network, IPv6Network
+
+from scapy.layers.l2 import Ether, Dot1Q
+from scapy.layers.inet import IP, TCP
+
+
+class TestABX(VppTestCase):
```
Comment (ondrej-fabry) [inline vpp/abx/vpp2202/test/test_abx.py:31]: Could you add a comment for the `TestABX` with a list of test cases we are testing here? To save time for other devs/testers needing to go through entire file.

**Review by ondrej-fabry — CHANGES_REQUESTED** [review]

File: vpp/abx/vpp2202/test/test_abx.py [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+        pkts,
+        output,
+        n_rx=None,
+        worker=None,
+        trace=True,
+        msg=None,
+        stats_diff=None,
+    ):
+        return super().send_and_expect(
+            intf, pkts, output, n_rx, worker, trace, msg, stats_diff
```
Comment (pt-maros) [inline vpp/abx/vpp2202/test/test_abx.py:257]: I have been getting following error. Removing `msg` and `stats_diff` parameters from method call fixed it.
```
Traceback (most recent call last):
  File "/home/maros/vpp/test/test_abx.py", line 277, in test_abx_basic_acl_mac_dst_sub_if
    pkts = self.send_and_expect(self.pg0, pkts, self.pg1)
  File "/home/maros/vpp/test/test_abx.py", line 256, in send_and_expect
    return super().send_and_expect(
TypeError: send_and_expect() takes from 4 to 7 positional arguments but 9 were given
```

**Review by pt-maros — COMMENTED** [review]
Would it be relevant to add a test for ACL rule with IPv6 prefix? Or a negative scenario when the packet doesn't get forwarded?

Nothing else comes to my mind when it comes to test scenarios. Beyond that, I noticed that the test file fails style check when `v22.02` branch is checked out. It passes it on `master` branch. Not sure if that's relevant, perhaps for some automated test in build service.

File: vpp/abx/vpp2202/test/test_abx.py (hunk 1 above)
Comment (PantheonVladimirB) [reply vpp/abx/vpp2202/test/test_abx.py:31]: Adding section with tests included.

File: vpp/abx/vpp2202/test/test_abx.py (hunk 2 above)
Comment (PantheonVladimirB) [reply vpp/abx/vpp2202/test/test_abx.py:257]: ```send_and_expect``` method fixed as mentioned above.

For checkstyle, decision is to use one from ```v22.02``` branch.

IPv6 tests are not priority at the moment, listed as TODO. Negative scenario could be implemented, but for now should be handled by ACL test cases since they determine which packets are to be forwarded.

**Review by ondrej-fabry — APPROVED** [review]
